internal/provider/cert_test.go

```diff
@@ -0,0 +1,77 @@
+package provider
```

Comment:
our compliance tools also seem to miss this copyright header at the top of this new file

```suggestion
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0

package provider
```